core/domain/topic_domain.py

```diff
@@ -133,6 +157,57 @@ def to_dict(self):
             'version': self.version
         }
 
+    def validate(self):
+        """Validates all properties of this topic and its constituents.
+
+        Raises:
+            ValidationError: One or more attributes of the Topic are not
+            valid.
+        """
+        if not isinstance(self.name, basestring):
+            raise utils.ValidationError(
+                'Expected name to be a string, received %s' % self.name)
+        if not isinstance(self.description, basestring):
+            raise utils.ValidationError(
+                'Expected description to be a string, received %s'
+                % self.description)
+
+        if not isinstance(self.language_code, basestring):
+            raise utils.ValidationError(
+                'Expected language code to be a string, received %s' %
+                self.language_code)
+        if not any([self.language_code == lc['code']
+                    for lc in constants.ALL_LANGUAGE_CODES]):
+            raise utils.ValidationError(
+                'Invalid language code: %s' % self.language_code)
+
+        if not isinstance(self.canonical_story_ids, list):
+            raise utils.ValidationError(
+                'Expected canonical story ids to be a list, received %s'
+                % self.canonical_story_ids)
+        for story_id in self.canonical_story_ids:
+            if story_services.get_story_by_id(story_id, False) is None:
+                raise utils.ValidationError(
+                    'The story with id %s doesn\'t exist' % story_id)
+
+        if not isinstance(self.additional_story_ids, list):
```

Comment:
Probably worth checking that all elements in these two story_ids lists are distinct, and that the lists have an empty intersection.

Reply:
Done.

Reply:
I don't see the check for "elements in each of the lists are distinct"?

Also you should have backend tests for both these cases.

Reply:
Oh, missed that part. Done, and added tests.